File: 48-RotateImage/solution.py
def rotate(self, matrix):
    n = len(matrix)

    # **Transpose the matrix**
    for i in range(n):
        for j in range(i, n):
            matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]

    # **Horizontal Flip**
    for row in matrix:
        row.reverse()  

File: 48-RotateImage/solution.js
function rotate(matrix) {
  let n = matrix.length;

  // Transpose the matrix
  for (let i = 0; i < n; i++) {
    for (let j = i; j < n; j++) {
      [matrix[i][j], matrix[j][i]] = [matrix[j][i], matrix[i][j]];
    }
  }

  // Horizontal Flip
  for (let row of matrix) {
    row.reverse();
  }
}
